common/source/docs/common-network.rst

```diff
@@ -0,0 +1,118 @@
+.. _common-network:
+
+===========================
+ArduPilot Network Interface
+===========================
+
+As of firmware 4.5 and later, ArduPilot provides a network interface framework to allow local and wide-area network connections. Several later generation autopilots, like Pixhawk6X and CubePilot CubeRed, provide Ethernet physical layer interfaces, which allows ArduPilot to connect to vehicle peripherals, data servers, and even the wide area network via IP using transport layer UDP or TCP protocols.
+
+Currently, hooks into the normal serial manager within ArduPilot now allow serial protocols not only to be connected to the autopilot via UART connections, but also via network connections. These are currently limited to using Ethernet physical layer.
+
+In addition, LUA Scripting bindings allow the autopilot to run network server type applications via LUA scripts. A web server LUA example is available as an applet that allows faster log downloads, as well as vehicle/system status data to be presented.
+
+For peripherals with a network interface ArduPilot has AP_Periph firmware that allows the easy development of network peripheral devices using the peripheral's Serial or DroneCAN interfaces to the peripheral device and an easily set up means to pass those devices' data to the autopilot over the network.
+
+ArduPilot Network Framework Overview
+====================================
+
+ArduPilot provides a NET manager facility that provides the IP protocol engine, as well as UDP and TCP transport layer engines. Four configurable IP ports are available, and each one can be a UDP or TCP, Client or Server. Data flow to/from these ports can be used by the autopilot as if the data streams were those of a:
+
+- A serial device supported by ArduPilot (ie. Rangefinder, Gimbal, serial ESC, etc.)
+- A MAVLink connection to another vehicle or GCS
+
+In addition, LUA Scripting running on an autopilot can access these data ports.
+
+.. image:: ../../../images/Net_framework.jpg
+
+Configuration
+=============
+Global ArduPilot Configuration
+------------------------------
+
+- :ref:`NET_ENABLED<NET_ENABLED>` must be set to "1" and parameters refreshed in order to see the following:
+
+- :ref:`NET_MACADDR0<NET_MACADDR0>`
+- :ref:`NET_MACADDR1<NET_MACADDR1>`
+- :ref:`NET_MACADDR2<NET_MACADDR2>`
+- :ref:`NET_MACADDR3<NET_MACADDR3>`
+- :ref:`NET_MACADDR4<NET_MACADDR4>`
+- :ref:`NET_MACADDR5<NET_MACADDR5>`
+
+This is the global 6 byte MAC address. The default is automatically generated from the autopilot's CPU unique id, but can be changed if desired.
+
+- :ref:`NET_IPADDR0<NET_IPADDR0>`
+- :ref:`NET_IPADDR1<NET_IPADDR1>`
+- :ref:`NET_IPADDR2<NET_IPADDR2>`
+- :ref:`NET_IPADDR3<NET_IPADDR3>`
+
+This is the autopilot's 4 byte IP address. It defaults to ``192.168.13.14``. It can be dynamically assigned if the network has a DHCP server by enabling ArduPilot's DHCP client using the :ref:`NET_DHCP<NET_DHCP>` parameter, or manually set as desired.
```

Comment:
It seems to be optional to set this. Tridge showed me how to setup ppp without setting NET_IPADDR[0123]. You might want to ask him how that's possible, so it can be explained here.

Reply:
see below